MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/query/FingerDrawQueryActivity.java

```diff
@@ -0,0 +1,391 @@
+  private List<Point> freehandTouchPointListForLine = new ArrayList<>();
+  private List<List<Point>> polygonList;
+  private Point firstScreenTouchPoint;
+  private boolean showSearchDataLocations = true;
+  private boolean drawSingleLineOnly = false;
+
+  private View.OnTouchListener customOnTouchListener = new View.OnTouchListener() {
+    @Override
+    public boolean onTouch(View view, MotionEvent motionEvent) {
+
+      LatLng latLngTouchCoordinate = mapboxMap.getProjection().fromScreenLocation(
+        new PointF(motionEvent.getX(), motionEvent.getY()));
+
+      Point screenTouchPoint = Point.fromLngLat(latLngTouchCoordinate.getLongitude(),
+        latLngTouchCoordinate.getLatitude());
+
+      // Draw the line on the map as the finger is dragged along the map
+      freehandTouchPointListForLine.add(screenTouchPoint);
+      drawLineSource = mapboxMap.getStyle().getSourceAs(FREEHAND_DRAW_LINE_LAYER_SOURCE_ID);
+      drawLineSource.setGeoJson(LineString.fromLngLats(freehandTouchPointListForLine));
+
+      // Draw a polygon area if drawSingleLineOnly == false
+      if (!drawSingleLineOnly) {
+        if (freehandTouchPointListForPolygon.size() == 0) {
+          firstScreenTouchPoint = screenTouchPoint;
+        }
+        if (freehandTouchPointListForPolygon.size() < 3) {
+          freehandTouchPointListForPolygon.add(screenTouchPoint);
+        } else if (freehandTouchPointListForPolygon.size() == 3) {
+          freehandTouchPointListForPolygon.add(screenTouchPoint);
+          freehandTouchPointListForPolygon.add(firstScreenTouchPoint);
+        } else {
+          freehandTouchPointListForPolygon.remove(freehandTouchPointListForPolygon.size() - 1);
+          freehandTouchPointListForPolygon.add(screenTouchPoint);
+          freehandTouchPointListForPolygon.add(firstScreenTouchPoint);
+        }
+
+        // Create and show a FillLayer polygon where the search area is
+        fillPolygonSource = mapboxMap.getStyle().getSourceAs(FREEHAND_DRAW_FILL_LAYER_SOURCE_ID);
+        polygonList = new ArrayList<>();
+        polygonList.add(freehandTouchPointListForPolygon);
+        fillPolygonSource.setGeoJson(Polygon.fromLngLats(polygonList));
+      }
+
+      // Take certain actions when the drawing is done
+      if (motionEvent.getAction() == MotionEvent.ACTION_UP) {
+
+        // If drawing polygon, add the first screen touch point to the end of
+        // the LineLayer list so that it's
+        if (!drawSingleLineOnly) {
+          freehandTouchPointListForLine.add(firstScreenTouchPoint);
+        }
+
+        if (showSearchDataLocations && !drawSingleLineOnly) {
+
+          // Use Turf to calculate the number of data points within a certain Polygon area
+          FeatureCollection pointsInSearchAreaFeatureCollection =
+            TurfJoins.pointsWithinPolygon(searchPointFeatureCollection,
+              FeatureCollection.fromFeature(Feature.fromGeometry(
+                Polygon.fromLngLats(polygonList))));
```

Comment:
This is already called a couple of lines above and the resulting object should be reused.

Reply:
Seems that I need to keep a `Polygon` variable at the very top of the example because it seems that the resulting object can't be given to `pointsWithinPolygon()`.

Reply:
I don't think that's the case. You can move the `drawnPolygon` declaration out of the https://github.com/mapbox/mapbox-android-demo/pull/983/files#diff-258a012908215cf52feb4555f609aa98R101.

Reply:
Yea, moved

```java
GeoJsonSource fillPolygonSource = mapboxMap.getStyle().getSourceAs(FREEHAND_DRAW_FILL_LAYER_SOURCE_ID);
      List<List<Point>> polygonList = new ArrayList<>();
      polygonList.add(freehandTouchPointListForPolygon);
      Polygon drawnPolygon = Polygon.fromLngLats(polygonList);
      fillPolygonSource.setGeoJson(drawnPolygon);
```